mne/channels/channels.py

```diff
@@ -69,17 +69,29 @@ def _contains_ch_type(info, ch_type):
         raise ValueError('`ch_type` is of class {actual_class}. It must be '
                          '`str`'.format(actual_class=type(ch_type)))
 
-    valid_channel_types = ('grad mag eeg stim eog emg ecg ref_meg resp '
-                           'exci ias syst seeg misc').split()
+    valid_channel_types = ('grad mag planar1 planar2 eeg stim eog emg ecg '
+                           'ref_meg resp exci ias syst seeg misc').split()
 
     if ch_type not in valid_channel_types:
         msg = ('The ch_type passed ({passed}) is not valid. '
-               'it must be {valid}')
+               'it must be one of [{valid}]')
         raise ValueError(msg.format(passed=ch_type,
-                                    valid=' or '.join(valid_channel_types)))
+                                    valid=', '.join(valid_channel_types)))
     return ch_type in [channel_type(info, ii) for ii in range(info['nchan'])]
 
 
+def _get_ch_type(info, ch_type):
+    """Helper to choose a singel channel type (usually for plotting)"""
+    if ch_type is None:
+        for type_ in ['mag', 'grad', 'planar1', 'planar2', 'eeg']:
+            if _contains_ch_type(info, type_):
+                ch_type = type_
+                break
+        else:
+            raise RuntimeError('No plottable channel types found')
+    return ch_type
```

Comment:
How are we supposed to know if channels are in the ICA object, then? I get this error now that I've changed it to use `ch_type in ica`:
```
  File "/home/larsoner/custombuilds/mne-python/mne/preprocessing/ica.py", line 1329, in plot_components
    image_interp=image_interp)
  File "/home/larsoner/custombuilds/mne-python/mne/viz/topomap.py", line 638, in plot_ica_components
    ch_type = _get_ch_type(ica, ch_type)
  File "/home/larsoner/custombuilds/mne-python/mne/channels/channels.py", line 92, in _get_ch_type
    if type_ in inst:
  File "/home/larsoner/custombuilds/mne-python/mne/channels/channels.py", line 152, in __contains__
    has_ch_type = _contains_ch_type(self.info, ch_type)
  File "/home/larsoner/custombuilds/mne-python/mne/channels/channels.py", line 81, in _contains_ch_type
    return ch_type in [channel_type(info, ii) for ii in range(info['nchan'])]
TypeError: 'NoneType' object has no attribute '__getitem__'
```
The bug should be fixed at the `ContainsMixin` level probably.